backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt

```diff
@@ -0,0 +1,109 @@
+package com.nosql.authservice.component.impl
+
+import com.nosql.authservice.common.exception.DataIntegrityViolationApplicationException
+import com.nosql.authservice.common.exception.enumerator.ApplicationExceptionCodeEnum
+import com.nosql.authservice.common.logger.logFailed
+import com.nosql.authservice.common.logger.logSuccess
+import com.nosql.authservice.component.UserComponent
+import com.nosql.authservice.entity.UserEntity
+import com.nosql.authservice.common.exception.ApplicationException
+import com.nosql.authservice.common.exception.NotFoundException
+import com.nosql.authservice.common.logger.logBefore
+import com.nosql.authservice.repository.UserRepository
+import kotlinx.coroutines.reactor.awaitSingle
+import com.nosql.authservice.common.logger.logger
+import com.nosql.authservice.dto.UserDto
+import com.nosql.authservice.exception.UserAlreadyExistsException
+import org.slf4j.Logger
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+import reactor.core.publisher.Mono
+
+@Component
+@Transactional
+class DefaultUserComponent(
+    private val userRepository: UserRepository,
+) : UserComponent {
+
+    private val logger: Logger by logger()
+
+    @Transactional
+    override suspend fun saveIfNotExists(user: UserEntity): UserEntity {
+        val operationDetails = "Save if not exists 'auth' record with login = "
+
+        logger.logBefore(operationDetails)
+
+        assertUserNotExists(user)
+
+        return userRepository.save(user)
+            .onErrorMap { handleError(it, operationDetails) }
+            .doOnSuccess { logger.logSuccess(operationDetails) }
+            .awaitSingle()
+    }
+
+    @Transactional(readOnly = true)
+    override suspend fun existsByLogin(login: String): Boolean {
+        val operationDetails = "Check if 'auth' record by login = '$login' exists"
+
+        logger.logBefore(operationDetails)
+
+        return userRepository.existsByLogin(login)
+            .onErrorMap { handleError(it, operationDetails) }
+            .doOnSuccess { logger.logSuccess(operationDetails) }
+            .awaitSingle()
+    }
+
+    @Transactional(readOnly = true)
```

Comment:
You could remove the `@Transactional` annotation from the class and add it only for write methods.